src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/TripSchedule.java

```diff
@@ -20,6 +22,8 @@
      */
     TripPattern getOriginalTripPattern();
 
+    LocalDate getServiceDate();
+
```

Comment:
I would like to point out this this duplicate the service calendar information. The service date is redundant, it can be found using the calendar info, trip and leg. We sometimes do duplicate information in the model, but be aware that that have a high maintenance cost. I would recommend retrieving this from the calendar service instead at the time of mapping into the API if it is not to difficult or error prune. We plan to clean up the transit layer, so maybe we can deal with it then.

Reply:
An alternative could be to have a `localDateFromStartOfTimeAndSecondsOffset()` function which may then be used for mapping to a `ServiceDate` ([commit](https://github.com/realcity/OpenTripPlanner/commit/4e9338d0f992a7bb59196ec45de1fac0d02510a6)):
```
leg.serviceDate = new ServiceDate(localDateFromStartOfTimeAndSecondsOffset(startOfTime, ((TripScheduleWithOffset) tripSchedule).getSecondsOffset()));
```

```
public static LocalDate localDateFromStartOfTimeAndSecondsOffset(ZonedDateTime startOfTime, int secondsOffset) {
    ZonedDateTime noon = startOfTime.plusSeconds(secondsOffset).plusHours(12);
    return noon.toLocalDate();
}
```

Would this be preferable?

Reply:
I think what you have done is ok for now, but I might refactor this as part of #3030. I will keep the `TripSchedule.getServiceDate()` method, but compute it instead of caching the value. 

I don't like "global" static methods, because you basically need to know they exist to use them and also the internal representation of the calendar leeks outside the model.